### ISSUE TITLE
Bug: type checking is not work with vue3 ref components

### ISSUE THREAD

**Issue by sklme** [typescript-eslint/typescript-eslint#5292]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

type checking is not work with vue3 ref components.
<img width="697" alt="image" src="https://user-images.githubusercontent.com/4933537/177074707-ed4808af-d6ba-49ec-b613-6b86509729d9.png">
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/4933537/177074718-e702a7df-7a9f-4806-9391-467fa61fda6b.png">

The type is ok in typescript, by fail in eslint. It should not produce any error.

### Reproduction Repository Link

https://github.com/sklme/vite-vue3-template

### Repro Steps

1) clone the repo
2) `pnpm install`
3) `npx eslint src/App.vue`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.30.4` |
| `@typescript-eslint/parser`            | `5.30.4` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.18` |
| `node`                                 | `18.2.0` |

**Comment by armano2** [comment]
this is actually to be expected as you are defining instance of component as any, `any + object = any`

```ts
declare module '*.vue' {
  import type { DefineComponent } from 'vue';
  const component: DefineComponent<{}, {}, any>;
  export default component;
}
```

in one of examples you can see that issue is not related to `ref` but rather to `T` 

```ts
type T = InstanceType<typeof TestExposeVue>; // T is any
```

this start to work perfectly fine when you provide `{}` instead of any in `DefineComponent`

```ts
const component: DefineComponent;
```

**Comment by armano2** [comment]
yes
